### ISSUE TITLE
Add aXe testing to automated Ci tests

### ISSUE THREAD

**Issue by nolanlawson** [nolanlawson/pinafore#513]
We can use this testcafe plugin: https://github.com/helen-dikareva/axe-testcafe

**Comment by sorin-davidoi** [comment]
The plugin uses an older version of aXe, but there are two pull requests opened if the latest version is needed.

**Comment by nolanlawson** [comment]
Interestingly it seems aXe cannot test Pinafore. I've tried `axe-cli` as well, and it just hangs.
![screenshot from 2018-12-05 22-53-07](https://user-images.githubusercontent.com/283842/49566786-adf9eb80-f8e0-11e8-946f-2c5da680a2f3.png)

I thought maybe it was due to CSP, but I disabled it and it still fails.

On the bright side, Lighthouse does seem to be working so perhaps this should be closed in favor of https://github.com/nolanlawson/pinafore/issues/746

**Comment by sorin-davidoi** [comment]
aXe runs for me in both Firefox Nightly and Chromium.

![image](https://user-images.githubusercontent.com/2109702/49657854-828a1480-fa38-11e8-8b9d-5b71d6e2e66e.png)

**Comment by nolanlawson** [comment]
axe works for me now, although it's finding some false positives: https://github.com/dequelabs/axe-core/issues/1176#issuecomment-473704254

**Comment by NickColley** [comment]
Want to get this going to ensure when we resolve contrast issues we dont accidentally regress so adding to milestone for WCAG compliance.

**Comment by nolanlawson** [comment]
@NickColley Thanks a lot. AA sounds totally achievable for us. (My understanding is that AAA is more like a "stretch goal" so not necessarily something we need to target?)

**Comment by NickColley** [comment]
Yeah some things you want to try and aim for AAA but sometimes it's actually impossible to meet all to AAA.

AA is what the main legislation for the EU and UK Public Sector Accessibility Regulations align with, and is becoming a sort of defacto baseline, and definitely achievable for us.

**Comment by NickColley** [comment]
Just a note on false positives, it's pretty common for axe to throw up issues that are false positives so what we'd do in this case is disable the rule that is being thrown and document why. This may also be a good way to introduce the tests as well, in a sort of 'failing test' situation where we add axe and raise github issues for each disabled rule.
